src/components/Explorer.js

```diff
@@ -64,20 +64,21 @@ class Explorer extends React.Component {
 
   componentDidMount() {
     const { appState, dispatch, router } = this.props
+    const { since, until } = appState.filters
+    const { query } = router.location
+
+    const clean = (val, alt) => {
+      const yr = +val
+      return yr >= 1960 && yr <= 2014 ? yr : alt
+    }
+
     const filters = {
       ...this.props.filters,
       ...router.params,
-      ...router.location.query,
+      since: clean(query.since, since),
```

Comment:
This could be a sort of weird behavior, since it means that if you have a working range, let's say 2000-2014 but try and select 1950-2014 this code looks like it will reset your range to 1960-2014. Is that correct? Might be better to just leave the value and use the previous one instead of forcing them to use one end of the range?